### ISSUE TITLE
Add option to display image of native numbers next to OCR results. 

### ISSUE THREAD

**Issue by jef-pearlman** [farkam135/GoIV#372]
If the OCR setup allows for it, I would love an option that, if turned on, displayed the graphical CP, HP, and point on the CP meter (don't know the proper word for it) next to the OCR'd result.  This would significantly streamline the process for people like me who want to check the data, since you wouldn't have to look up and down twice to compare and remember to check the HP _before_ you trigger GoIV, as the GoIV UI blocks the native HP display.  So the second GoIV line would look something like:

CP 304  [Image of 384]        HP 48  [Image of 48]

Note that this issue is exacerbated by #237 and the lack of a warning when OCR fails, though it would still be useful for ensuring that OCR did not "succeed" but produce an incorrect result. 

Thanks!

**Comment by Blaisorblade** [comment]
> as the GoIV UI blocks the native HP display.

😢 that's very unfortunate, we tried fixing it but it's layout dependent (it's better for me on master than it was, not sure if that changed since 3.1). But your idea would avoid that and is generally interesting.

It might be technically nontrivial (sending images inside intent seems bad and maybe to be avoided, not sure), but that's hopefully solvable.

> and point on the CP meter (don't know the proper word for it) next to the OCR'd result.

I guess you mean the dot on the arc? We do show the dot on the dot. Are you asking for a snapshot of the actual dot we OCR? That's harder since we don't do that—we look at points going backward till we find a white one. IIUC the coordinates we look at are somewhere inside the right dot, but not necessarily centered. And IIUC changing this is lots of trial and error on lots of snapshots.

The other things we OCR are the candy and pokemon name (and at some point, the amount of candy for evolution, since that and the candy name determines the pokemon uniquely). For that we assume the correct pokemon name is obvious to the user.

> Note that this issue is exacerbated by #237 and the lack of a warning when OCR fails

Do you actually mean #237? That's about being "scrolled down". However, I agree that displaying "10" for a failure is not a good idea, we should just leave the field empty.

**Comment by rpgrca** [comment]
> I guess you mean the dot on the arc? We do show the dot on the dot. Are you asking for a snapshot of the actual dot we OCR? 

I think he says he wants the numeric level the dot represents, like L18.5. Maybe @jef-pearlman can upload a screenshot to see how he's currently seeing it?

**Comment by jef-pearlman** [comment]
Yeah--after writing this, I noticed the layout dependency (the HP view is not blocked on another device I have).

On the second point, I did mean the dot on the arc.  For that, I was envisioning a small snapshot around where the detected (red) dot is places, so that in the normal case one can see a bit of the white portion of the arc to the left of the dot, the gray portion to the right, and the red dot on top of the white dot, to quickly verify it's in the right place.  I think having a coordinate somewhere inside the dot would be enough, since the rectangle would be large enough to include the dot and a bit of the arc regardless of where exactly inside the dot it was detected.

I agree the name probably doesn't need this feature, and that 10 shouldn't be used for OCR failures.  (I personally favor a popup warning for failures, but that's me.)  

Hope that clarifies.  Happy to upload a screenshot if there's still any ambiguity.

And a random thought: is it really OCR if there's no scanning involved?  [R]asterCR?  Mostly kidding; I know everyone uses OCR for this type of usage.

**Comment by nahojjjen** [comment]
I personally do not think this feature is something that should be added, as it would require us to add and maintain placing images in the input layout and designing it to work well. I think it creates more problems than it solves.

**Comment by sarav** [comment]
@nahojjjen This is a constant issue on my phone. Every time I scan the pokemon, it opens up in appraisal mode that ends up blocking CP and HP. I'll have to fold appraisal mode to see what the HP is (which is barely missed from being blocked) and then open appraisal again and go through the rest of the steps. Adding this feature would make the steps a lot more simpler.

**Comment by Blaisorblade** [comment]
@sarav #432 won't get fixed if nobody opens it—I was literally looking for it yesterday. I've done it for you. (A chat is not a bug tracker 😉 ).

**Comment by nahojjjen** [comment]
Can we close this issue as wontfix?

**Comment by Blaisorblade** [comment]
I'll go with yes. Doing this in a decent-looking way on different devices is rather hard. It's much simpler to leave the underlying values visible. We fixed #237 and now the appraisal box is closed again by default, and that was much easier.
